pkg/indexer/indexer.go

```diff
@@ -157,17 +160,17 @@ type builtStreamer struct {
 func configureLogStream(
 	ctx context.Context,
 	builder *blockchain.RpcLogStreamBuilder,
-	cfg config.ContractsOptions,
+	reader blockchain.AppChainReader,
+	maxChainDisconnectTime time.Duration,
 	querier *queries.Queries,
 ) (*builtStreamer, error) {
-	messagesTopic, err := buildMessagesTopic()
+	messagesAddress, err := reader.ContractAddress(blockchain.EventTypeMessageSent)
 	if err != nil {
 		return nil, err
 	}
-
 	messagesTracker, err := NewBlockTracker(
 		ctx,
-		cfg.AppChain.GroupMessageBroadcasterAddress,
+		messagesAddress,
 		querier,
 	)
```

Comment:
I noticed that we are fetching the latest block number as a separate call every time we read from the blockchain, and that we're also doing DB writes of the latest block number every time we write to the database. Does this fall under the existing optimization work you were tackling @fbac?

I chose to abstract at the chain reader level rather than the whole indexer, so that even when we are using the database-backed chain reader, we are still reusing as much of the existing indexer code as possible. It does mean that we potentially inherit any performance limitations, but on the flipside any performance fixes accrue to both implementations as well.